File: Hijeck/app/src/main/java/com/example/hijeck/ShowingAlldataActivity.java
package com.example.hijeck;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.SearchView;

import android.database.Cursor;
import android.os.Bundle;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.ArrayAdapter;
import android.widget.ListAdapter;
import android.widget.ListView;
import android.widget.Toast;

import java.util.ArrayList;

public class ShowingAlldataActivity extends AppCompatActivity {

    DataBaseClassAdding db;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_showing_alldata);

        ListView listView = (ListView) findViewById(R.id.ListView);
        db = new DataBaseClassAdding(this);

        Cursor data = db.getAllData();
        UserSecretAdapter adapter = new UserSecretAdapter(this, R.layout.adapterview);

        if (data.getCount() == 0) {
            Toast.makeText(getApplicationContext(), "YOU HAVE NO SECRETS PLEASE ADD", Toast.LENGTH_SHORT).show();
        } else {
            while (data.moveToNext()) {

                UserSecrets us = new UserSecrets(data.getString(0), data.getString(1), data.getString(2));
                adapter.add(us);
            }
            listView.setAdapter(adapter);
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.searchmaenu,menu);
        getMenuInflater().inflate(R.menu.mymenu, menu);
        MenuItem searchitem=menu.findItem(R.id.search_menu);

        SearchView searchView= (SearchView) searchitem.getActionView();
        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String query) {
                return false;
            }

            @Override
            public boolean onQueryTextChange(String newText) {
                return false;
            }
        });
        return true;

    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        int id = item.getItemId();

        switch (id) {
            case R.id.home:
                break;
            case R.id.logout:
                Toast.makeText(getApplicationContext(), "LOGOUT", Toast.LENGTH_SHORT).show();
            case R.id.delete:
                db.cleardata();
                Toast.makeText(getApplicationContext(), "You do not have any secrets", Toast.LENGTH_SHORT).show();
                break;

        }
        return true;
    }
}

File: GoogleSummerOfCode/app/src/main/java/com/example/googlesummerofcode/ShowingAlldataActivity.kt
package com.example.googlesummerofcode

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.ListView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.SearchView

class ShowingAlldataActivity : AppCompatActivity() {
    var db: DataBaseClassAdding? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_showing_alldata)
        val listView =
            findViewById<View>(R.id.ListView) as ListView
        db = DataBaseClassAdding(this)
        val data = db!!.allData
        val adapter = UserSecretAdapter(this, R.layout.adapterview)
        if (data!!.count == 0) {
            Toast.makeText(
                applicationContext,
                "YOU HAVE NO SECRETS PLEASE ADD",
                Toast.LENGTH_SHORT
            ).show()
        } else {
            while (data!!.moveToNext()) {
                val us =
                    UserSecrets(data!!.getString(0), data!!.getString(1), data!!.getString(2))
                adapter.add(us)
            }
            listView.adapter = adapter
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.searchmaenu, menu)
        menuInflater.inflate(R.menu.mymenu, menu)
        val searchitem = menu.findItem(R.id.search_menu)
        val searchView =
            searchitem.actionView as SearchView
        searchView.setOnQueryTextListener(object :
            SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(query: String): Boolean {
                return false
            }

            override fun onQueryTextChange(newText: String): Boolean {
                return false
            }
        })
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.itemId
        when (id) {
            R.id.home -> {
            }
            R.id.logout -> {
                Toast.makeText(applicationContext, "LOGOUT", Toast.LENGTH_SHORT).show()
                db!!.cleardata()
                Toast.makeText(
                    applicationContext,
                    "You do not have any secrets",
                    Toast.LENGTH_SHORT
                ).show()
            }
            R.id.delete -> {
                db!!.cleardata()
                Toast.makeText(
                    applicationContext,
                    "You do not have any secrets",
                    Toast.LENGTH_SHORT
                ).show()
            }
        }
        return true
    }
}